### ISSUE TITLE
Handling error status codes when a body is present

### ISSUE THREAD

**Issue by yoadsn** [azure-contrib/node-azure-search#31]
I was wondering what would be the most robust way to handle the following situation:
- Calling a an API endpoint feature with the wrong (too old) API version
- Getting back a body which is JSON parseable - for example:
```
{
    "Message": "No HTTP resource was found that matches the request URI 'https://xxx.search.windows.net/synonymmaps?api-version=2016-09-01'."
}
```
- Response code is 404

With today's implementation, this is not considered an error response since:
a - there is no `error` property of the returned JSON body [here](https://github.com/azure-contrib/node-azure-search/blob/942a8509d3a1006fbd77dd717115a5fefefcccca/index.js#L412)
b - a JSON body exists. [here](https://github.com/azure-contrib/node-azure-search/blob/942a8509d3a1006fbd77dd717115a5fefefcccca/index.js#L64-L69)

And the body is passed as a valid response to the callback.

I am not sure how standard such a reply is, but it's part of the API surface of Azure Search and so we do have to handle that IMO. (documented [here](https://docs.microsoft.com/en-us/rest/api/searchservice/http-status-codes))

My suggestion is:
- Validate status codes are within 200-207 regardless of the existence of a response body
- If status codes fall outside this range - merge into the response body (or empty object as a fallback) the `code` property.

Would love to place a PR for that if that solution makes sense.

**Comment by richorama** [comment]
@yoadsn excellent suggestion, and I would definitely merge a PR 👍 

On a slight tangent, I wonder whether we should bump up the default API version, as the library supports features available in `2016-09-01-Preview`?

**Comment by yoadsn** [comment]
@richorama see other PR.
Reg. the API version - I have set it as the default for the tests suite.
Still, my feeling is that it should remain opt-in until the features are out of preview.

The reasoning is partly because if you create/update an index (for example) using the new API version - old version calls will stop working for that object.
I suspect some users might upgrade and discover they can no longer use their old scripts / revert back without deleting stuff and recreating. ouch.

**Comment by richorama** [comment]
implemented by #32
